### PR TITLE
fixed create table async query bug

### PR DESCRIPTION
**Pull request by paulstn**
### Description
fixed create table async query bug, also added content to show a success without schema and also failure

https://github.com/opensearch-project/dashboards-query-workbench/assets/25872706/1047ef2d-ea71-44a5-80fe-74e01682e181

Update to failure portion:

https://github.com/opensearch-project/dashboards-query-workbench/assets/25872706/909c7d6b-34fc-4ad2-b2b2-33b49cb9a3b8

 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/QueryResults/AsyncQueryBody.tsx [hunk 1]
```diff
@@ -3,30 +3,42 @@
...
+          </EuiText>
+          <EuiText>error: {asyncQueryError}</EuiText>
+        </>
+      ) : (
+        <>
+          <EuiLoadingSpinner size="l" />
+          <EuiText>
+            <h3>Query running</h3>
+          </EuiText>
+          <EuiText>status: {asyncLoadingStatus}</EuiText>
```
Comment (Swiddis) [inline public/components/QueryResults/AsyncQueryBody.tsx:38]: Should `status` here be caplitalized?

File: public/components/Main/main.tsx [hunk 2]
```diff
@@ -511,7 +520,7 @@ export class Main extends React.Component<MainProps, MainState> {
         this.setState({
           queries: queries,
           queryResults: [result],
-          queryResultsTable: resultTable,
+          queryResultsTable: result.data['schema'].length > 0 ? resultTable : [],
```
Comment (Swiddis) [inline public/components/Main/main.tsx:523]: Should this have unit tests of some sort?

File: public/components/QueryResults/AsyncQueryBody.tsx [hunk 3]
```diff
@@ -3,30 +3,42 @@
  * SPDX-License-Identifier: Apache-2.0
  */
```
Comment (ps48) [inline public/components/QueryResults/AsyncQueryBody.tsx:5]: As this is a new file introduced in 2.11, can you please follow the naming convention where: file name is all lower case and separated by underscore. `async_query_body.tsx`

Comment (paulstn) [reply public/components/QueryResults/AsyncQueryBody.tsx:5]: modified

Comment (ps48) [reply public/components/QueryResults/AsyncQueryBody.tsx:5]: Doc: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/DEVELOPER_GUIDE.md#filenames

File: public/components/Main/main.tsx (hunk 2 above)
Comment (paulstn) [reply public/components/Main/main.tsx:525]: coming later

**Review by ps48 — APPROVED** [review]

**Pull request merged**